R/get-template.R

```diff
@@ -0,0 +1,179 @@
+#' @title Get metadata template keys
+#'
+#' @description Get the keys from a metadata template stored as an excel or csv
+#' or registered as a JSON schema in Synapse.
+#'
+#' @param synID Synapse ID of an excel or csv file containing a metadata
+#'   template
+#' @param id The id of JSON metadata schema registered in Synapse or Synapse ID
+#' of an excel or csv file containing a metadata template
+#' @inheritParams get_synapse_annotations
+#' @param ... Additional arguments passed to syn$get()
+#' @return Character vector of template column names
+#' @export
+#' @examples
+#' \dontrun{
+#' syn <- synapse$Synapse()
+#' syn$login()
+#' get_template(synID = "syn12973252", syn = syn)
+#' get_template(id = "syn12973252", syn = syn)
+#' }
+get_template <- function(id = NA, syn, synID = NA, ...) {
+  ## Use synID if given
+  if (!is.na(synID)) {
+    return(get_template_keys_synID(syn = syn, synID = synID, ...))
+  }
+  if (!is.na(id)) {
+    ## Check if id was used to give a synID
+    if (grepl("^syn", id)) {
+      return(get_template_keys_synID(syn = syn, synID = id, ...))
+    }
+    ## If not a synID, assume a schema id
```

Comment:
What is the schema id convention?

Reply:
organization.name-schema.name-version

For example, the schema in the default config is:
https://github.com/Sage-Bionetworks/dccvalidator/blob/8d7b75c88a2f0b7ee76640d2d22ef3fbcd4dacdb/inst/config.yml#L28

This is registered under the organization sysbio.metadataTemplates with the schema name assay.rnaSeq. There is no version, but if there was, it would be something like `sysbio.metadataTemplates-assay.rnaSeq-0.0.1`, similar to versions in the synapseAnnotations repository.

More info can be found in confluence for [Synapse schema syntax](https://sagebionetworks.jira.com/wiki/spaces/PLFM/pages/989167828/Schemas+for+extending+Synapse+Objects#Name-spacing).